opentelemetry-sdk/src/metrics/meter_provider.rs

```diff
@@ -145,34 +145,14 @@ impl Drop for SdkMeterProviderInner {
     }
 }
 impl MeterProvider for SdkMeterProvider {
-    fn versioned_meter(
-        &self,
-        name: &'static str,
-        version: Option<&'static str>,
-        schema_url: Option<&'static str>,
-        attributes: Option<Vec<KeyValue>>,
-    ) -> Meter {
+    fn library_meter(&self, scope: InstrumentationScope) -> Meter {
         if self.inner.is_shutdown.load(Ordering::Relaxed) {
             return Meter::new(Arc::new(NoopMeter::new()));
         }
 
-        let mut builder = Scope::builder(name);
-
-        if let Some(v) = version {
-            builder = builder.with_version(v);
-        }
-        if let Some(s) = schema_url {
-            builder = builder.with_schema_url(s);
-        }
-        if let Some(a) = attributes {
-            builder = builder.with_attributes(a);
-        }
-
-        let scope = builder.build();
-
         if let Ok(mut meters) = self.inner.meters.lock() {
             let meter = meters
-                .entry(scope)
+                .entry(InstrumentationScope::clone(&scope))
```

Comment:
No need to clone here.

```suggestion
                .entry(scope)
```

Reply:
fixed